x-pack/plugins/fleet/public/components/package_policy_actions_menu.tsx

```diff
@@ -174,7 +174,7 @@ export const PackagePolicyActionsMenu: React.FunctionComponent<{
       {isEnrollmentFlyoutOpen && (
         <EuiPortal>
           <AgentEnrollmentFlyout
-            agentPolicy={agentPolicy}
+            agentPolicy={agentPolicies.length === 1 ? agentPolicies[0] : undefined} // in case of multiple policies, show the selector in the flyout
```

Comment:
in the case of multiple agent policies, could we still preselect the first agent policy but allow others to be selected?

also, is there a way to restrict the options to just the agent policies that integration policy belongs to?

Reply:
> could we still preselect the first agent policy but allow others to be selected?
The undefined on this line only tells the flyout to show the agent policy selector, then the user can select what they want.

>  is there a way to restrict the options to just the agent policies that integration policy belongs to?
I did this change in https://github.com/elastic/kibana/pull/193532/commits/c8d26035ec95e6feeee81acdf1b623bc22d4668e 
I recorded a short video to show the updated behavior:

https://github.com/user-attachments/assets/7f0786b0-029c-4003-b075-fa50ff4129a3